dpc/src/account/tests.rs

```diff
@@ -55,6 +56,17 @@ mod testnet1 {
         assert_eq!(ALEO_TESTNET1_ADDRESS, address.to_string());
     }
 
+    #[test]
+    fn test_private_key_to_decryption_key() {
+        // Attempt to sample for a new account ITERATIONS times.
+        for _ in 0..ITERATIONS {
+            let private_key = PrivateKey::<Testnet1Parameters>::new(&mut thread_rng());
+            let decryption_key = private_key.to_decryption_key().unwrap();
+            // Enforce the MSB of the scalar field element is 0 by convention.
+            assert_eq!(Some(&false), decryption_key.to_bits_be().iter().next());
```

Comment:
HOLD. 

I found a potential bug in my implementation of `to_bits_be`. The truncation is performed starting from the MSB direction, but it should be from the LSB direction.

https://github.com/AleoHQ/snarkVM/blob/testnet2/fields/src/fp_256.rs#L408

Will post a fix, likely to this branch.

Reply:
Great catch, will hold

Reply:
And then, it is also curious why this test did not fail.

Reply:
Perhaps the implementation is correct because `BigInteger` does the following:
```rs
            /// Returns `self` as a boolean array in big-endian order, without leading zeros.
            fn to_bits_be(&self) -> Vec<bool> {
                BitIteratorBE::new(self).collect::<Vec<_>>()
            }
```

Reply:
That makes sense. Although it is more concerning, now it is not fixed length, but we have used this in a few places where it is needed.

Let me use `resize` to make sure it is the right length.

Reply:
Oh wait, I think that implementation is wrong, it should be:
```rs
            /// Returns `self` as a boolean array in big-endian order, without leading zeros.
            fn to_bits_be(&self) -> Vec<bool> {
                BitIteratorBE::new_without_leading_zeros(self).collect::<Vec<_>>()
            }
```

Reply:
ok, it seems resizing is not needed, and we may want to change that comment in `BigInteger`, as the leading zeros are still there.

Reply:
There are two commits related to this that you will need to review:

https://github.com/AleoHQ/snarkVM/pull/347/commits/671982fb5d339cb18da0a1d43e88bed6f63a3770
https://github.com/AleoHQ/snarkVM/pull/347/commits/eb8f36f29be2882f29ee728afa23ced57867c132